ghost/core/test/e2e-webhooks/pages.test.js

```diff
@@ -0,0 +1,122 @@
+const {
+    agentProvider,
+    mockManager,
+    fixtureManager,
+    matchers
+} = require('../utils/e2e-framework');
+const {
+    stringMatching,
+    anyUuid,
+    anyLocalURL,
+    anyISODateTime,
+    anyObjectId,
+    anyContentVersion,
+    anyNumber
+} = matchers;
+
+const anyGhostAgent = stringMatching(
+    /Ghost\/\d+\.\d+\.\d+\s\(https:\/\/github.com\/TryGhost\/Ghost\)/
+);
+
+const tierSnapshot = {
+    id: anyObjectId,
+    created_at: anyISODateTime,
+    updated_at: anyISODateTime
+};
+
+const buildAuthorSnapshot = (roles = false) => {
+    const authorSnapshot = {
+        last_seen: anyISODateTime,
+        created_at: anyISODateTime,
+        updated_at: anyISODateTime
+    };
+
+    // NOTE: this is such a bad hack! for the reasons I did not investigate the "add" event does not include
+    //       the roles but the "published" does! massive inconsistency and needs to be fixed one day
+    if (roles) {
```

Comment:
The 'roles' here does not seem to ever be "true", so probably no need for this whole block here